src/manifold/src/csg_tree.cpp

```diff
@@ -419,34 +457,16 @@ std::vector<std::shared_ptr<CsgNode>> &CsgOpNode::GetChildren(
     bool finalize) const {
   auto impl = impl_.GetGuard();
   auto &children_ = impl->children_;
-  if (children_.empty() || (impl->simplified_ && !finalize) || impl->flattened_)
-    return children_;
-  impl->simplified_ = true;
```

Comment:
Happy to see this simplification, but what was the difference between `simplified` and `flattened`?

Reply:
`simplified` did not perform flattening if the nodes are shared, iirc.

Reply:
Flattening (also aliased here to finalization) was forcing leaf nodes, while simplification was flattening 1 level of op nodes nesting (adopting the children of same op children, and in the case of difference, of union op children past the first child).

Have now renamed finalize & flattened_ to force[d]ToLeafNodes[_] to disambiguate, and simplified this method further.